code/modules/antagonists/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,1093 @@
+	. = ..()
+	if(!target)
+		return
+	if(target.stat)
+		to_chat(usr, span_warning("[target] must be conscious!"))
+		revert_cast()
+		return
+	if(IS_SHADOW_OR_THRALL(target))
+		to_chat(usr, span_warning("You cannot glare at allies!"))
+		revert_cast()
+		return
+	var/mob/living/carbon/human/M = target
+	usr.visible_message(span_warning("<b>[usr]'s eyes flash a purpleish-red!"))
+	var/distance = get_dist(target, usr)
+	if (distance <= 2)
+		target.visible_message("<span class='danger'>[target] suddendly collapses...</span>")
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and you lose control of your movements!</span>")
+		target.Paralyze(10 SECONDS)
+		M.silent += 10
+	else //Distant glare
+		var/loss = 100 - (distance * 10)
+		target.adjustStaminaLoss(loss)
+		target.stuttering = loss
+		to_chat(target, span_userdanger("A purple light flashes across your vision, and exhaustion floods your body..."))
+		target.visible_message(span_danger("[target] looks very tired..."))
+	charge_counter = 0
+	start_recharge()
+	remove_ranged_ability()
+	usr = null
+	target = null
+
+/obj/effect/proc_holder/spell/aoe_turf/veil //Puts out most nearby lights except for flares and yellow slime cores
+	name = "Veil"
+	desc = "Extinguishes most nearby light sources."
+	panel = "Shadowling Abilities"
+	charge_max = 150 //Short cooldown because people can just turn the lights back on
+	human_req = TRUE
+	clothes_req = FALSE
+	range = 5
+	action_icon_state = "veil"
+	action_icon = 'icons/mob/actions.dmi'
+	var/admin_override = FALSE ///Requested by Shadowlight213. Allows anyone to cast the spell, not just shadowlings.
+
+/obj/effect/proc_holder/spell/aoe_turf/proc/extinguishItem(obj/item/I, cold = FALSE) //Does not darken items held by mobs due to mobs having separate luminosity, use extinguishMob() or write your own proc.
+	var/blacklisted_lights = list(/obj/item/flashlight/flare, /obj/item/flashlight/slime)
+	if(istype(I, /obj/item/flashlight))
+		var/obj/item/flashlight/F = I
+		if(F.on)
+			if(!cold)
+				return
+			if(is_type_in_list(F, blacklisted_lights))
+				F.visible_message(span_warning("The sheer cold shatters [F]!"))
+				qdel(F)
+			if(is_type_in_list(I, blacklisted_lights))
+				I.visible_message(span_userdanger("[I] dims slightly before scattering the shadows around it."))
+				return F.on //Necessary because flashlights become 0-luminosity when held.  I don't make the rules of lightcode.
+			F.on = FALSE
+			F.update_brightness()
+	else if(istype(I, /obj/item/pda))
+		var/obj/item/pda/P = I
+		P.set_light_on(FALSE)
+		P.update_appearance()
+		P.update_action_buttons(force = TRUE)
+	I.set_light(0)
+	return I.luminosity
+
+/obj/effect/proc_holder/spell/aoe_turf/proc/extinguishMob(mob/living/H, cold = FALSE)
+	for(var/obj/item/F in H)
+		if(cold)
+			extinguishItem(F, TRUE)
+		extinguishItem(F)
+
+/obj/effect/proc_holder/spell/aoe_turf/veil/cast(list/targets,mob/user = usr)
```

Comment:
In the future, this will 100% have to use light_eater, i'll let look to have it test merged beforehand though.

Reply:
Ok